pkg/templates/machinecontroller/deployment.go

```diff
@@ -765,6 +765,8 @@ func machineControllerDeployment(cluster *kubeoneapi.KubeOneCluster, credentials
 
 		args = append(args, "-node-hyperkube-image", hyperkubeImage)
 		args = append(args, "-node-kubelet-repository", poseidonKubeletImage)
+
+		args = append(args, "-node-pause-image", pauseImage)
```

Comment:
when `if s.Cluster.RegistryConfiguration == nil || s.Cluster.RegistryConfiguration.OverwriteRegistry == "" {` is true, this pauseImage will be empty string, and machine-controller probably will blowup, this flag should be added under condition `if pauseImage != ""`.

Reply:
If that condition is true, `if cluster.RegistryConfiguration != nil && cluster.RegistryConfiguration.OverwriteRegistry != "" {` on L762 would be false, so we wouldn't set the `-node-pause-image` flag (or any other relevant flag). I don't think that we need additional check.